src/drivers/px4fmu/fmu.cpp

```diff
@@ -1340,6 +1340,15 @@ PX4FMU::cycle()
 
 				actuator_outputs_s actuator_outputs = {};
 				actuator_outputs.timestamp = hrt_absolute_time();
+
+				for (unsigned i = 0; i < actuator_controls_s::NUM_ACTUATOR_CONTROL_GROUPS; i++) {
+					actuator_outputs.timestamp_sample = _controls[i].timestamp_sample;
```

Comment:
I'm not sure about this. There are other actuator_controls publications in the system that might be throwing this off. https://github.com/PX4/Firmware/blob/master/src/modules/fw_att_control/fw_att_control_main.cpp#L1233

For now only using actuator_controls_0 should be sufficient if in the VTOL case you always put the timestamp_sample coming from the active controller.